docs/docsite/rst/dev_guide/developing_modules_best_practices.rst

```diff
@@ -1,197 +1,144 @@
+.. _developing_modules_best_practices:
 .. _module_dev_conventions:
 
+*****************************************
 Conventions, Best Practices, and Pitfalls
-`````````````````````````````````````````
+*****************************************
 
-As a reminder from the example code above, here are some basic conventions
-and guidelines:
+.. contents:: Topics
+   :local:
 
-* If the module is addressing an object, the parameter for that object should be called 'name' whenever possible, or accept 'name' as an alias.
+As you design and develop modules, follow these basic conventions and best practices:
 
-* If you have a company module that returns facts specific to your installations, a good name for this module is `site_facts`.
+Scoping your module(s)
+======================
 
-* Modules accepting boolean status should generally accept 'yes', 'no', 'true', 'false', or anything else a user may likely throw at them.  The AnsibleModule common code supports this with "type='bool'".
+Especially if you want to contribute your module(s) back to Ansible Core, make sure each module includes enough logic and functionality, but not too much. If you're finding these guidelines tricky, consider :ref:`whether you really need to write a module <module_dev_should_you>` at all.
```

Comment:
Drop the "especially" or rearrange the sentence so it flows better, e.g., "Make sure each module includes enough logic and functionality, but not too much — especially if you want to contribute your module(s) back to Ansible Core."